src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -221,7 +269,7 @@ public void CheckVersionWithTestHost()
 
             protocolNegotiated.Set();
         };
-        _channel.MessageReceived += _onMessageReceived;
+        _channel.MessageReceived += onMessageReceived;
```

Comment:
Should you be calling `TrySetupMessageReceiver` here?

Reply:
currently this message receiver does not have an aborted callback, which is why I left it as is
and I didn't want to change the semantics of the handshake for the aborted scenario in this PR